cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -1207,7 +1238,7 @@ func (csr *ClusterStateRegistry) GetScaleUpFailures() map[string][]ScaleUpFailur
 	defer csr.Unlock()
 	result := make(map[string][]ScaleUpFailure)
 	for nodeGroupId, failures := range csr.scaleUpFailures {
-		result[nodeGroupId] = failures
+		result[nodeGroupId] = append(result[nodeGroupId], failures...)
```

Comment:
Creating a new slice here because I think it'd add another layer of protection in case the caller mutates the return value (`result` variable).